### PR TITLE
RDKTV-16030  Addressed class hierarchy review comments

### PR DESCRIPTION
**Pull request by tamilselvanak**

### CODE REVIEW

**Review by apatel859 — CHANGES_REQUESTED** [review]
Could you setup a meeting including Anand K, Ramkumar, Deeksheet and me to finalize on this

**Comment by tamilselvanak** [comment]
Hi @mhughesacn,
Sure will append those lines in top level NOTICE.

File: ControlSettings/ControlSettingsTV.h [hunk 1]
```diff
@@ -17,27 +17,36 @@
...
+        ~ControlSettingsTV();
+
+    public:
+        //   IPlugin methods
+        // -------------------------------------------------------------------------------------------------------
+        const std::string Initialize(PluginHost::IShell* service);
+        void Deinitialize(PluginHost::IShell* service);
+        std::string Information() const;
+        void AddRef() const;
+        uint32_t Release() const;
```
Comment (apatel859) [inline ControlSettings/ControlSettingsTV.h:48]: check on this if required

File: ControlSettings/ControlSettings.cpp [hunk 2]
```diff
@@ -248,29 +239,13 @@ namespace Plugin {
...
-        devicePtr = new STB;
-#else
-        devicePtr = new TV;
-#endif
 
 	ControlSettings::_instance = this;
 	InitializeIARM();
 
-        if(devicePtr->isTvSupportEnabled())//TV Specific API
-	{
```
Comment (apatel859) [inline ControlSettings/ControlSettings.cpp:261]: Are we doing unregister?

File: ControlSettings/ControlSettingsSTB.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+ "locator":"libControlSettingsSTB.so",
```
Comment (apatel859) [inline ControlSettings/ControlSettingsSTB.json:2]: check if remove STB

File: ControlSettings/ControlSettingsSTB.cpp [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+       LOGINFO("Exit\n");
+    }
+
+    void ControlSettingsSTB::Deinitialize()
+    {
+       LOGINFO("Entry\n");
+       LOGINFO("Exit\n");
+    }
+
+    std::string ControlSettingsSTB::getErrorString (tvError_t eReturn)
```
Comment (apatel859) [inline ControlSettings/ControlSettingsSTB.cpp:53]: tv should be not be there in stb

**Review by ddevad — APPROVED** [review]

**Review by apatel859 — APPROVED** [review]

**Pull request merged**